File: src/main/java/lan/dk/podcastserver/manager/worker/downloader/NoOpDownloader.java
package lan.dk.podcastserver.manager.worker.downloader;

import lan.dk.podcastserver.entity.Item;

/**
 * Created by kevin on 10/03/2016 for Podcast Server
 */
public class NoOpDownloader implements Downloader {
    @Override
    public Item download() {
        return null;
    }

    @Override
    public Downloader setItem(Item item) {
        return null;
    }

    @Override
    public Item getItem() {
        return null;
    }

    @Override
    public String getItemUrl() {
        return null;
    }

    @Override
    public void startDownload() {}

    @Override
    public void pauseDownload() {}

    @Override
    public void stopDownload() {}

    @Override
    public void finishDownload() {}

    @Override
    public void resetDownload() {}

    @Override
    public Integer compatibility(String url) {
        return null;
    }

    @Override
    public void run() {}
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/download/downloaders/noop/NoOpDownloader.kt
package com.github.davinkevin.podcastserver.download.downloaders.noop

import com.github.davinkevin.podcastserver.download.ItemDownloadManager
import com.github.davinkevin.podcastserver.download.downloaders.Downloader
import com.github.davinkevin.podcastserver.download.downloaders.DownloaderFactory
import com.github.davinkevin.podcastserver.download.downloaders.DownloadingInformation
import com.github.davinkevin.podcastserver.download.downloaders.DownloadingItem


class NoOpDownloaderFactory: DownloaderFactory {
    override fun with(information: DownloadingInformation, itemDownloadManager: ItemDownloadManager): Downloader {
        return NoOpDownloader(information, itemDownloadManager)
    }

    override fun compatibility(downloadingInformation: DownloadingInformation) = -1
}

class NoOpDownloader(
    val state: DownloadingInformation,
    val manager: ItemDownloadManager,
) : Downloader {

    override val downloadingInformation: DownloadingInformation
        get() = state

    override fun download(): DownloadingItem = state.item
    override fun startDownload() = failDownload()
    override fun stopDownload() {}
    override fun finishDownload() {}
    override fun failDownload() = manager.removeACurrentDownload(state.item.id)
}
